client/src/views/workspaces/WorkspacesPage.vue

```diff
@@ -423,6 +430,43 @@ async function openWorkspace(workspace: WorkspaceInfo): Promise<void> {
   }
 }
 
+async function renameWorkspace(workspace: WorkspaceInfo, newName: WorkspaceName): Promise<void> {
+  const result = await parsecRenameWorkspace(newName, workspace.id);
+  if (result.ok) {
+    await refreshWorkspacesList();
+    informationManager.present(
+      new Information({
+        message: translate('WorkspacesPage.RenameWorkspaceModal.success', { newName: newName }),
+        level: InformationLevel.Success,
+      }),
+      PresentationMode.Toast,
+    );
+  } else {
+    informationManager.present(
+      new Information({
+        message: translate('WorkspacesPage.RenameWorkspaceModal.error'),
+        level: InformationLevel.Error,
+      }),
+      PresentationMode.Toast,
+    );
+  }
+}
+
+async function openRenameWorkspaceModal(workspace: WorkspaceInfo): Promise<void> {
+  const newWorkspaceName = await getTextInputFromUser({
+    title: translate('WorkspacesPage.RenameWorkspaceModal.pageTitle'),
+    trim: true,
+    validator: workspaceNameValidator,
+    inputLabel: translate('WorkspacesPage.RenameWorkspaceModal.label'),
+    placeholder: translate('WorkspacesPage.RenameWorkspaceModal.placeholder'),
+    okButtonText: translate('WorkspacesPage.RenameWorkspaceModal.rename'),
```

Comment:
I would add a selection and a defaultValue, similar to what's done with files. See https://github.com/Scille/parsec-cloud/blob/d4fdf118be3c83e19b63750338dd6b79c132aa4c/client/src/views/files/FoldersPage.vue#L632.